### PR TITLE
workspace oriented dependencies

### PR DESCRIPTION
**Pull request by heemankv**
# This PR resolves issue #99

Going with this convention for PR : 
- If a dependencies is only being used in a single crate and it's specific to that crate then no need to move it to workspace dependency.
- If a dependency is being used in multiple crates but with different features then it should be defined as a workspace dependency and individual features must be pulled in each crate.
- If any dependency is being used in many crates without any `ifs and buts` then we define it as a workspace dependency.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -50,13 +54,13 @@ url = { version = "2.5.0", features = ["serde"] }
 uuid = { version = "1.7.0", features = ["v4", "serde"] }
 httpmock = { version = "0.7.0", features = ["remote"] }
 num-bigint = { version = "0.4.4" }
-arc-swap = { version = "1.7.1" }
+arc-swap = { version = "1.7.0" }
```
Comment (heemankv) [inline Cargo.toml:57]: `arc-swap` should be removed after PR #107  get's merged.

File: crates/settlement-clients/ethereum/Cargo.toml [hunk 2]
```diff
@@ -4,18 +4,18 @@ version.workspace = true
 edition.workspace = true
 
 [dependencies]
-alloy = { workspace = true, features = ["full", "node-bindings"] }
 alloy-primitives = { version = "0.7.7", default-features = false }
+
+alloy = { workspace = true, features = ["full", "node-bindings"] }
```
Comment (heemankv) [inline crates/settlement-clients/ethereum/Cargo.toml:9]: Wanted to update alloy to latest version, but it's specifically required with varied features in the crates. Left it as it is for now.
Screenshot 2024-09-06 at 12 38 53 PM

**Comment by heemankv** [comment]
Completed Self Review, looks good!

File: crates/settlement-clients/ethereum/Cargo.toml (hunk 2 above)
Comment (apoorvsadana) [reply crates/settlement-clients/ethereum/Cargo.toml:9]: are you sure, it's possible that we just copied the import from somewhere and maybe the both work on the latest version?

File: crates/da-clients/ethereum/Cargo.toml [hunk 3]
```diff
@@ -15,14 +15,14 @@ alloy = { git = "https://github.com/alloy-rs/alloy", rev = "68952c0", features =
...
 ] }
 async-trait = { workspace = true }
-c-kzg = "1.0.0"
+c-kzg = { workspace = true }
 color-eyre = { workspace = true }
 da-client-interface = { workspace = true }
-dotenv = "0.15"
-mockall = "0.12.1"
-reqwest = { version = "0.12.3" }
+dotenvy.workspace = true
```
Comment (apoorvsadana) [inline crates/da-clients/ethereum/Cargo.toml:21]: Nitpick: should we change everything to one format? either `{ workspace = true }` or `.workspace = true`?

**Review by apoorvsadana — APPROVED** [review]

**Pull request merged**